netket/hilbert/homogeneous.py

```diff
@@ -72,10 +72,9 @@ def __init__(
         self._is_finite = local_states is not None
 
         if self._is_finite:
-            self._local_states = np.asarray(local_states)
-            assert self._local_states.ndim == 1
-            self._local_size = self._local_states.shape[0]
-            self._local_states = self._local_states.tolist()
+            self._local_states = local_states
+            self._local_size = len(self._local_states)
+            # TODO make local_states a HashablAarray and remove _local_states_frozen
```

Comment:
Reminder for a followup: we should enforce that `local states` must be a StaticRange and use that for comparisons in `_attr`. This allows us to remove this ugly `_local_states_frozen`.

Reply:
yeah, this is just a reminder for the future. Not for this PR.